node/_fs/_fs_read.ts

```diff
@@ -0,0 +1,113 @@
+// Copyright 2018-2021 the Deno authors. All rights reserved. MIT license.
+import { Buffer } from "../buffer.ts";
+import { assert } from "../../testing/asserts.ts";
+import { ERR_INVALID_ARG_VALUE } from "../_errors.ts";
+
+type readOptions = {
+  buffer: Buffer | Uint8Array;
+  offset: number;
+  length: number;
+  position: number | null;
+};
+
+type BinaryCallback = (
+  err: Error | null,
+  bytesRead: number | null,
+  data?: Buffer,
+) => void;
+type Callback = BinaryCallback;
+
+export function read(fd: number, callback: Callback): void;
+export function read(
+  fd: number,
+  options: readOptions,
+  callback: Callback,
+): void;
+export function read(
+  fd: number,
+  buffer: Buffer | Uint8Array,
+  offset: number,
+  length: number,
+  position: number | null,
+  callback: Callback,
+): void;
+export function read(
+  fd: number,
+  optOrBuffer?: Buffer | Uint8Array | readOptions | Callback,
+  offsetOrCallback?: number | Callback,
+  length?: number,
+  position?: number | null,
+  callback?: Callback,
+): void {
+  let cb: Callback | undefined;
+  let offset: number = 0,
+    buffer: Buffer | Uint8Array;
+
+  if (length == null) {
+    length = 0;
+  }
+
+  if (typeof offsetOrCallback === "function") {
+    cb = offsetOrCallback;
+  } else if (typeof optOrBuffer === "undefined") {
```

Comment:
I think this should be `typeof optOrBuffer === "function"`, and the test case doesn't seem catching this error